modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -20,9 +20,10 @@ const (
 	ICAPrefix string = "ics-27"
 )
 
-// GenerateAddress returns a truncated SHA256 hash using the provided port string
-func GenerateAddress(port string) []byte {
-	return tmhash.SumTruncated([]byte(port))
+// GenerateAddress returns an sdk.AccAddress using the provided port identifier and derived from the IBC interchainaccounts module account
+func GenerateAddress(portID string) sdk.AccAddress {
```

Comment:
Is it worth adding a test for this?

Reply:
Yeah for sure, I can add one with the other updates! Should be pretty basic